### PR TITLE
closure for new async calls

### PR DESCRIPTION
**Pull request by matei-p**

### CODE REVIEW

**Review by andrei-marinica — DISMISSED** [review]

File: arwen/elrondapi/managedei.go [hunk 1]
```diff
@@ -614,14 +621,45 @@ func v1_5_managedCreateAsyncCall(
...
+		extraGasForCallback,
+		callbackClosure)
+}
+
+//export v1_5_managedGetCallbackClosure
+func v1_5_managedGetCallbackClosure(
+	context unsafe.Pointer,
+	callbackClosureHandle int32,
+) {
+	host := arwen.GetVMHost(context)
```
Comment (sasurobert) [inline arwen/elrondapi/managedei.go:645]: gas use ?

**Review by sasurobert — CHANGES_REQUESTED** [review]

File: arwen/elrondapi/managedei.go (hunk 1 above)
Comment (matei-p) [reply arwen/elrondapi/managedei.go:659]: fixed

**Review by sasurobert — DISMISSED** [review]

File: arwen/contexts/async.go [hunk 2]
```diff
@@ -986,6 +994,29 @@ func (context *asyncContext) HasLegacyGroup() bool {
...
+
+// GetCallbackClosure gets the async call callback closure
+func (context *asyncContext) GetCallbackClosure() ([]byte, error) {
+	if context.callbackParentCall == nil {
+		stackContext := context.Clone()
+		stackContext, err := stackContext.(*asyncContext).loadParentContextFromStackOrStorage()
+		if err != nil {
+			return nil, arwen.ErrAsyncNoCallbackForClosure
+		}
+		context.callbackParentCall = stackContext.(*asyncContext).
```
[the remark this answers is not in the document]
Comment (matei-p) [reply arwen/contexts/async.go:1010]: We need it. Certain scenarios will fail otherwise

[the remark this answers is not in the document]
Comment (sasurobert) [reply arwen/contexts/async.go:1010]: smae for line 1006 - should work without any type assertion.

**Review by sasurobert — APPROVED** [review]

**Review by sasurobert — APPROVED** [review]

**Pull request merged**